j2cl-maven-plugin/src/main/java/com/vertispan/j2cl/mojo/AbstractBuildMojo.java

```diff
@@ -255,6 +255,7 @@ private Project buildProject(MavenProject mavenProject, Artifact artifact, boole
                         throw new ProjectBuildingException(p.getId(), "Failed to resolve this project's artifact file", e);
                     }
 
+                    getLog().info("Building dependency "+p.getGroupId()+":"+p.getArtifactId()+":"+p.getVersion());
```

Comment:
So it doesn't help your infinite loop problem, but there is already debug logging at line 99 that draws a rough tree diagram of the "dependency graph"

Reply:
i think you can remove this? or at the very least demote it to debug, but then we still have duplicate log lines with line 99

Reply:
If the build hangs, it won't print the tree later on. Removed it.